lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -331,22 +333,75 @@ private JourneyResponse getJourneyResponse(
     private VerifiableCredentialBundle getVerifiableCredentials(
             String userId, String evcsAccessToken)
             throws CredentialParseException, EvcsServiceException {
-        if (configService.enabled(EVCS_READ_ENABLED)) {
+
+        var tacticalVcs = verifiableCredentialService.getVcs(userId);
+
+        if (configService.enabled(EVCS_WRITE_ENABLED) || configService.enabled(EVCS_READ_ENABLED)) {
             var vcs =
                     evcsService.getVerifiableCredentialsByState(
-                            userId, evcsAccessToken, CURRENT, PENDING_RETURN);
-            var pendingReturnVcs = vcs.get(PENDING_RETURN);
-            // use pending return vcs to determine identity if available
-            if (!isNullOrEmpty(pendingReturnVcs)) {
-                return new VerifiableCredentialBundle(pendingReturnVcs, true, true);
+                            userId, evcsAccessToken, CURRENT, PENDING_RETURN, PENDING);
+
+            if (configService.enabled(EVCS_WRITE_ENABLED)) {
+                logIdentityMismatches(tacticalVcs, vcs);
             }
-            var currentVcs = vcs.get(CURRENT);
-            if (!isNullOrEmpty(currentVcs)) {
-                return new VerifiableCredentialBundle(currentVcs, true, false);
+            if (configService.enabled(EVCS_READ_ENABLED)) {
+                var pendingReturnVcs = vcs.get(PENDING_RETURN);
+                // use pending return vcs to determine identity if available
+                if (!isNullOrEmpty(pendingReturnVcs)) {
+                    return new VerifiableCredentialBundle(pendingReturnVcs, true, true);
+                }
+                var currentVcs = vcs.get(CURRENT);
+                if (!isNullOrEmpty(currentVcs)) {
+                    return new VerifiableCredentialBundle(currentVcs, true, false);
+                }
             }
         }
-        return new VerifiableCredentialBundle(
-                verifiableCredentialService.getVcs(userId), false, false);
+        return new VerifiableCredentialBundle(tacticalVcs, false, false);
+    }
+
+    private void logIdentityMismatches(
+            List<VerifiableCredential> tacticalVcs,
+            Map<EvcsVCState, List<VerifiableCredential>> evcsEvcs) {
+
+        // if we have pending vcs just check those
+        var evcsToCheck = evcsEvcs.get(PENDING);
+        if (!isNullOrEmpty(evcsToCheck)) {
+            evcsToCheck = evcsEvcs.get(CURRENT);
+        }
+        // filter out unmigrated vcs from the tactical vcs
+        tacticalVcs = tacticalVcs.stream().filter(vc -> vc.getMigrated() != null).toList();
+
+        // check all the selected evcs vcs are in the tactical store
+        if (hasIdentityMismatch(evcsToCheck, tacticalVcs)) {
```

Comment:
I'm not sure if I have this the right way round, do we:
 - check all the tactical vcs exist in the pending/current evcs
 - check all the pending/current evcs exist in the tactical store
 - both